### PR TITLE
Add LLEXT support for AArch64

### PR DESCRIPTION
**Pull request by awojasinski**
Introducing support for LLEXT on Arch64 architecture. The patch includes:
- arch-specific extension linking
- extension of targets for `samples/subsys/llext/modules`
- alignment of llext core source code for AArch64
- minor cleanup of symbol definition for architecture specific relocation codes (now in arch specific code)

This PR is based on
- #77826

### CODE REVIEW

**Comment by awojasinski** [comment]
I'll convert it into "ready for review" once #77826 is merged

**Review by pillo79 — COMMENTED** [review]
This looks amazing, thank you! :tada: :+1:

**Comment by awojasinski** [comment]
rebased on newest revision of #77826

**Review by npitre — DISMISSED** [review]

File: tests/subsys/llext/simple/testcase.yaml
Comment (pillo79) [inline tests/subsys/llext/simple/testcase.yaml:40]: The current `readonly` test intentionally avoids all memory protection issues so that they can be checked separately, and I'd rather not change that. 
I suggest we add a specific test for targets that strictly need MMU support, as it was made clear in the linked PR discussion:
```
llext.simple.readonly_mmu:
    arch_allow: arm64
    filter: CONFIG_MMU
    integration_platforms:
      - qemu_cortex_a53         # ARM Cortex-A53 (ARMv8-A ISA)
 ```
 @teburd what do you think?

**Review by pillo79 — COMMENTED** [review]
CI is green, great job! :rocket: Just one thing I'd like to confirm.

File: tests/subsys/llext/simple/testcase.yaml
Comment (teburd) [reply tests/subsys/llext/simple/testcase.yaml:40]: seems reasonable to me

**Review by pillo79 — COMMENTED** [review]
LGTM (once the A9 issue in CI is fixed) :+1:

**Review by npitre — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Review by pillo79 — APPROVED** [review]

**Review by lyakh — APPROVED** [review]

**Pull request merged**
